docs/adrs/0010-jetstream-shared-schema.md

```diff
@@ -0,0 +1,155 @@
+# Jetstream/Nimbus Shared Schema Architecture
+
+* Status: proposed
+* Deciders: Mike Williams
+* Date: 2023-04-21
+
+Technical Story: https://mozilla-hub.atlassian.net/browse/EXP-2265
+
+## Context and Problem Statement
+
+Experiment analysis for the Nimbus experimentation platform occurs in a separate project called Jetstream. Jetstream delivers results to Experimenter, which must parse these results based on the expected format of Jetstream’s output. However, Experimenter is forced to make assumptions about what it expects in the results, increasing the potential for buggy behavior during both ingestion and on the Results UI for Experimenter. Currently, various schemas are defined in a few different places for different purposes, but one schema definition shared between Experimenter and Jetstream would allow for validation of outputs from Jetstream while removing the need for Experimenter to make assumptions about the data it is receiving from Jetstream. This document describes the decisions around how and where that schema will be defined.
+
+
+## Decision Drivers
+
+* Single schema definition shared by Experimenter and Jetstream
+* No functional changes to Jetstream workflow
+  * Jetstream can (and should) use the schema definition in forming/validating outputs
+* Process for making schema changes
+* Simplify Experimenter ingestion of analysis results
+
+
+## Considered Options
+
+The schema definition and its location were related but independent decisions, so their respective options are enumerated here separately.
+
+### Schema Definition
+
+* [**Option 1**] Pydantic
+* [**Option 2**] Typescript
+
+### Schema Location
+
+* [**Option A**] `jetstream` repository
+* [**Option B**] `experimenter` (Nimbus) repository
+* [**Option C**] `nimbus-shared` repository
```

Comment:
Wasn't there some talk around having a `nimbus` repository at some point where `experimenter` and the new nimbus web effort will live? Would it make sense to have schemas in there as well (instead of moving them into experimenter)?

Reply:
Yes, I originally had a note about this but it appears that I accidentally dropped it in editing. But the important point is that the `experimenter` repo _is_ the (planned) `nimbus` repo, so your thought here is both valid and in line with the ADR.